src/Service/DashboardHelper.php

```diff
@@ -12,186 +12,184 @@
                 'count' => $boardNewPartySubmissionCount,
+                'button_style' => 'secondary',
             ];
 
             // On agenda
             $boardAgendaUrl = $this->router->generate('case_index', ['case_filter' => [
-                'board' => $boardFilterOption,
+                'board' => $board->getId(),
                 'specialStateFilter' => CaseSpecialFilterStatuses::ON_AGENDA,
             ]]);
 
             $boardAgendaCount = $this->caseRepository->findCountOfCasesWithActiveAgendaByBoard($board);
 
             $boardAgendaData = [
+                'label' => $this->translator->trans('On agenda', [], 'dashboard'),
                 'url' => $boardAgendaUrl,
                 'count' => $boardAgendaCount,
+                'button_style' => 'info',
             ];
 
             // Has exceeded one or more deadlines
             $boardExceededDeadlineUrl = $this->router->generate('case_index', ['case_filter' => [
-                'board' => $boardFilterOption,
+                'board' => $board->getId(),
                 'deadlines' => CaseDeadlineStatuses::SOME_DEADLINE_EXCEEDED,
             ]]);
 
             $boardExceededDeadlineCount = $this->caseRepository->findCountOfCasesWithSomeExceededDeadlineByBoard($board);
 
             $boardExceededDeadlineData = [
+                'label' => $this->translator->trans('Deadline reached', [], 'dashboard'),
                 'url' => $boardExceededDeadlineUrl,
                 'count' => $boardExceededDeadlineCount,
+                'button_style' => 'dark',
             ];
 
-            array_push($gridInformation, [
-                'grid_label' => $board->getName(),
-                'hearing_data' => $boardHearingData,
-                'new_party_submission_data' => $boardNewPartySubmissionData,
-                'agenda_data' => $boardAgendaData,
-                'exceeded_deadline_data' => $boardExceededDeadlineData,
+            $boardCount = $boardHearingCount + $boardNewPartySubmissionCount + $boardAgendaCount + $boardExceededDeadlineCount;
+
+            array_push($boardsInformation, [
+                'label' => $board->getName(),
+                'count' => $boardCount,
+                'rows' => [
+                    $boardHearingData,
+                    $boardNewPartySubmissionData,
+                    $boardAgendaData,
+                    $boardExceededDeadlineData,
+                ],
             ]);
         }
 
-        return $gridInformation;
-    }
-
-    private function getCurrentUserFilterOption(User $user): int
-    {
-        $caseworkers = $this->userRepository->findByRole('ROLE_CASEWORKER', ['name' => 'ASC']);
-
-        // Reindex caseworkers as UserRepository findByRole does not guarantee incrementing indexes starting from 0
-        $reindexedCaseworkers = array_values($caseworkers);
-
-        $userKey = array_search($user, $reindexedCaseworkers);
-
-        return $userKey;
+        return $boardsInformation;
```

Comment:
You don't need `$boardsInformation`. Just return 

```php
[
 'label' => $board->getName(),
 'count' => $boardCount,
 'rows' => …
]
```

for consistency with `getUserColumnCaseInformation`.

Reply:
Removed multiple unnecessary variables and used the `$rows` approach. This also made adding a potential row a lot simpler.